src/proto/types.rs

```diff
@@ -72,44 +72,83 @@ pub struct CloseCode(NonZeroU16);
+    pub const INVALID_FRAME_PAYLOAD_DATA: Self = Self(match NonZeroU16::new(1007) {
+        Some(val) => val,
+        None => unreachable!()
+    });
     /// Endpoint is terminating the connection because it has received a message
     /// that violates its policy.
-    pub const POLICY_VIOLATION: Self = Self(unsafe { NonZeroU16::new_unchecked(1008) });
+    pub const POLICY_VIOLATION: Self = Self(match NonZeroU16::new(1008) {
+        Some(val) => val,
+        None => unreachable!()
+    });
     /// Endpoint is terminating the connection because it has received a message
     /// that is too big for it to process.
-    pub const MESSAGE_TOO_BIG: Self = Self(unsafe { NonZeroU16::new_unchecked(1009) });
+    pub const MESSAGE_TOO_BIG: Self = Self(match NonZeroU16::new(1009) {
+        Some(val) => val,
+        None => unreachable!()
+    });
     /// Client is terminating the connection because it has expected the server
     /// to negotiate one or more extension, but the server didn't return them in
     /// the response message of the WebSocket handshake.
-    pub const MANDATORY_EXTENSION: Self = Self(unsafe { NonZeroU16::new_unchecked(1010) });
+    pub const MANDATORY_EXTENSION: Self = Self(match NonZeroU16::new(1010) {
+        Some(val) => val,
+        None => unreachable!()
+    });
     /// Server is terminating the connection because it encountered an
     /// unexpected condition that prevented it from fulfilling the request.
-    pub const INTERNAL_SERVER_ERROR: Self = Self(unsafe { NonZeroU16::new_unchecked(1011) });
+    pub const INTERNAL_SERVER_ERROR: Self = Self(match NonZeroU16::new(1011) {
+        Some(val) => val,
+        None => unreachable!()
+    });
     /// Service is restarted. A client may reconnect, and if it choses to do,
     /// should reconnect using a randomized delay of 5--30s.
-    pub const SERVICE_RESTART: Self = Self(unsafe { NonZeroU16::new_unchecked(1012) });
+    pub const SERVICE_RESTART: Self = Self(match NonZeroU16::new(1012) {
+        Some(val) => val,
+        None => unreachable!()
+    });
     /// Service is experiencing overload. A client should only connect to a
     /// different IP (when there are multiple for the target) or reconnect to
     /// the same IP upon user action.
-    pub const SERVICE_OVERLOAD: Self = Self(unsafe { NonZeroU16::new_unchecked(1013) });
+    pub const SERVICE_OVERLOAD: Self = Self(match NonZeroU16::new(1013) {
+        Some(val) => val,
+        None => unreachable!()
+    });
     /// The server was acting as a gateway or proxy and received an invalid
     /// response from the upstream server. This is similar to the HTTP 502
     /// status code.
-    pub const BAD_GATEWAY: Self = Self(unsafe { NonZeroU16::new_unchecked(1014) });
+    pub const BAD_GATEWAY: Self = Self(match NonZeroU16::new(1014) {
+        Some(val) => val,
+        None => unreachable!()
+    });
```

Comment:
The optimizer will easily prove that the const value is not zero. While this change makes the code bigger I'd see it as in preparation for [`Option::unwrap`](https://doc.rust-lang.org/1.83.0/std/option/enum.Option.html#method.unwrap) becoming const evaluatable in v1.83. Also removes noise when searching the codebase for `unsafe`.

Reply:
Also fun fact: clippy just added a lint about this https://rust-lang.github.io/rust-clippy/master/index.html#useless_nonzero_new_unchecked

Reply:
This comment is outdated after https://github.com/Gelbpunkt/tokio-websockets/pull/95#discussion_r1929808876.